libc/src/search/lfind.cpp

```diff
@@ -0,0 +1,29 @@
+//===-- Implementation of lfind   -------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/search/lfind.h"
+#include "src/__support/CPP/cstddef.h" // cpp::byte
+#include "src/__support/common.h"
+#include "src/__support/macros/config.h"
+
+namespace LIBC_NAMESPACE_DECL {
+LLVM_LIBC_FUNCTION(void *, lfind,
+                   (void *key, void *base, size_t *nmemb, size_t size,
+                    int (*compar)(void *, void *))) {
+  cpp::byte *next = reinterpret_cast<cpp::byte *>(base);
+  cpp::byte *end = next + (*nmemb * size);
```

Comment:
Usually with interfaces that accept the number of members and a size, there's a potential for `*nmemb * size` to overflow, which should be checked by the library function.  We should do that here.

We seem to have a `mul_overflow` helper fn in __support/memory_size.h, perhaps that should be moved to __support/math_extras.h with the others, then used here?

Reply:
The [standard](https://pubs.opengroup.org/onlinepubs/9799919799/functions/lfind.html) says we should return `nullptr` if an "error" occurs. 

I'll modify this to use `mul_overflow` to calculate the byte length, if it overflows, I'll return `nullptr` indicating an error.

Then in a subsequent PR, we can consider moving `mul_overflow` into `__support/math_extras.h`.
That way we can update all the `#include`s at once.

Reply:
This is resolved by [23d045e](https://github.com/llvm/llvm-project/pull/114692/commits/23d045e50e007a53ba966d0e771c124983150f9f)